lib/core/src/Cardano/Wallet.hs

```diff
@@ -622,10 +624,10 @@ restoreBlocks ctx wid blocks nodeTip = db & \DBLayer{..} -> do
     logCheckpoint cp = logInfo tr $
         "Creating checkpoint at " <> pretty (currentTip cp)
 
-    logDelegation :: (SlotId, PoolId) -> IO ()
-    logDelegation (slotId, poolId) = logInfo tr $ mconcat
+    logDelegation :: (SlotId, DelegationCertificate) -> IO ()
+    logDelegation (slotId, cert) = logInfo tr $ mconcat
         [ "Discovered delegation to pool "
-        , pretty poolId
+        , pretty (dlgCertPoolId cert)
```

Comment:
Is it possible for this expression to evaluate to `Nothing`? If so, then under which circumstances?

Reply:
I think it is correct as it follows from : 
```
dlgCertPoolId :: DelegationCertificate -> Maybe PoolId
dlgCertPoolId = \case
    CertDelegateNone{} -> Nothing
    CertDelegateFull _ poolId -> Just poolId
```

Reply:
Exactly, this has been introduced in 0.7.5 if I recall correctly so, we better use it instead of a hacky hack.

Reply:
Yet the log line won't look the best here, I'll update.